### DISCUSSION TITLE
ARXML checker/checking

### DISCUSSION THREAD

**Discussion by nithinvelupur** [girishchandranc/autosarfactory#27] [Q&A]
Is there any checking/validation that is done during export or import of ARXML? I am looking to create DEXT ARXML file and would like to ensure that the files i create are in-line with the AUTOSAR DEXT specification document.

PS: Sorry to highlight this as an issue. Didn't find the 'discussions' option for this repo.

**Comment by girishchandranc** [comment]
@nithinvelupur Currently, there is no explicit schema validation done. But, the model is 1:1 representation of the schema and hence if required, you could validate the output arxml file against the required schema. 

I know people are using autosarfactory to create the DEXT extract and so it works.

**Comment by nithinvelupur** [comment]
> @nithinvelupur Currently, there is no explicit schema validation done. But, the model is 1:1 representation of the schema and hence if required, you could validate the output arxml file against the required schema.
> 
> I know people are using autosarfactory to create the DEXT extract and so it works for the people.

______
Thanks. Will post an issue, IF any.

**Comment by girishchandranc** [comment]
@nithinvelupur  sure.. Hope this tool helps you. Please feel free to post issue if you find. Thanks!

**Comment by nithinvelupur** [comment]
NP. I am currently unable to find any 'check' rules for such ARXML schema validation. Do you know where exactly I can get that?
I will try to contribute to your repo with possibly a checker module.

If that's OK with you.

**Reply by girishchandranc** [reply]
@nithinvelupur  sure.. that's fine. The autosar schema files can be downloaded from https://www.autosar.org/search?tx_solr%5Bfilter%5D%5B0%5D=category%3AR23-11&tx_solr%5Bq%5D=. 

You could use lxml(https://lxml.de/validation.html) to validate the generated arxml(from autosarfactory) against the required autosar schema. 

If you wish to contribute to the repo, then I recommend that you must place the autosar schema for each of the versions in the subdirectory under https://github.com/girishchandranc/autosarfactory/tree/main/autosar_releases and call the checker script based on the autosar version.

Ps: lxml is already added as a dependency to this project and hence you can directly use it.

btw, I moved this issue to discussion.
